android/app/src/main/java/com/zulipmobile/MainActivity.kt

```diff
@@ -50,24 +58,19 @@ open class MainActivity : ReactActivity() {
     /* Returns true just if we did handle the intent. */
     private fun maybeHandleIntent(intent: Intent?): Boolean {
         // We handle intents from "sharing" something to Zulip.
-        if (intent?.action == Intent.ACTION_SEND) {
-            handleSend(intent)
-            return true
+        if (intent != null) {
+            if ((intent.action == Intent.ACTION_SEND) or (intent.action == Intent.ACTION_SEND_MULTIPLE)) {
+                handleSend(intent)
+                return true
+            }
         }
-        // TODO also handle ACTION_SEND_MULTIPLE?
-        //   See: https://developer.android.com/training/sharing/receive#receiving-data-activity
 
         // For other intents, let RN handle it.  In particular this is
         // important for VIEW intents with zulip: URLs.
         return false
     }
 
     private fun handleSend(intent: Intent) {
-        // Intent is reused after quitting, skip it.
-        if ((getIntent().flags and Intent.FLAG_ACTIVITY_LAUNCHED_FROM_HISTORY) != 0) {
-            return;
-        }
-
```

Comment:
Hmm interesting. Good catch!

    sharing: Fix bug when app is launched from history.
    
    This was required because of the following bug:
    1. share from any app to zulip.
    2. cancel the share.
    3. return back to zulip by selecting it from minimized apps.
    4. now go back to the previous app used to share to zulip.
    5. share something to zulip again.
    6. the sharing screen will not be visible.
    
    Intended behavior: Sharing screen should be visible in step 6
    
    This bug was possibly introduced in:
    
    70e9ad9d47ae2e192558cb4caa9ae8781251b926
    
    The placement of the code addressing the issue described in that commit
    was erraneous, as it was causing send intents coming from `onNewIntent`
    method to be ignored.

I think the bug in that commit is actually subtler. It's that this function is doing the rest of its work on `intent`, the parameter it was passed... but this conditional is instead looking at `getIntent()`, a getter on the `Activity`, which is giving it the intent the activity was originally started with.

What if you instead just change this conditional to refer to `intent` rather than `getIntent()`? Does that also resolve the symptoms?

Even if it does, it might be a good refactor anyway to move this conditional to `onCreate` -- it really should only be relevant there. NB that if we do, then it should still be edited to say `intent`, for a related reason -- it should match what we're passing to `maybeHandleIntent` just below that.

It might be clearest to have two commits, actually: one that fixes the bug, by saying `intent` here instead of `getIntent()`, and then optionally a second one that moves this to `onCreate`.

Reply:
> What if you instead just change this conditional to refer to intent rather than getIntent()? Does that also resolve the symptoms?

Yup, This also resolves the issue. I have now made two commits as suggested.